### ISSUE TITLE
Support for smart invert on iOS has regressed

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#120930]
Sometime back in 2018 I wired up support for smart invert by apply an inversion filter to our images when this setting is toggled. However on recent testing, smart invert decides not to invert anything in a Flutter. Thus the inversion has the opposite effect.

If this is no longer a reasonable choice, we should remove this setting and the corresponding Paint.invertColors flag.

@chunhtai @goderbauer

**Comment by jmagman** [comment]
https://github.com/flutter/engine/pull/6176
https://github.com/flutter/flutter/pull/21457

**Comment by jonahwilliams** [comment]
Its possible this is a recent change, or may not be related to an iOS version. On the iPhone 7 this still works as expected.

**Comment by jmagman** [comment]
@jonahwilliams if you give repro steps and expected results we can ask the triagers to try on different versions and devices.

**Comment by jonahwilliams** [comment]
Steps:

1. go to the iOS settings and search for "Smart Invert" (not classic invert) and enable it.
2. flutter run the old gallery in flutter/dev/integration_tests/flutter_gallery
3. most colors should be inverted.
4. Navigate to studies, then "fortnightly"
5. The image header should _not_ be inverted.

This works for me on iPhone 7. On the iPad pro after enabling smart invert the screen of the flutter app is not inverted at all

**Comment by jmagman** [comment]
The app was inverted and the fortnightly image header was not inverted (so it still worked) on my iPhone 6s running iOS 13.7, iPhone 12 Pro running iOS 15.4.1, and on my iPhone 14 running iOS 16.2.

All the simulator devices and iOS versions I tried did reproduce the issue and the app wasn't inverted, but the header image was.

**Comment by jonahwilliams** [comment]
Okay, something weird was going on. Closing
